### PR TITLE
`metabase.xrays` API namespace

### PR DESCRIPTION
**Pull request by camsaul**
- Consolidate `metabase.transforms`, `metabase.domain-entities`, and `metabase.automagic-dashboards` into a single `metabase.xrays` module. 
- Remove unused `/api/transform` endpoint.
- Some additional `dev.deps-graph` helpers.

### CODE REVIEW

File: dev/src/dev/deps_graph.clj [hunk 1]
```diff
@@ -56,3 +56,66 @@
...
+               (let [module-deps (get module->deps module)]
+                 (not-empty (into (sorted-set)
+                                  (filter (fn [dep]
+                                            (circular-dependency? module dep)))
+                                  module-deps))))]
+       (into (sorted-map)
+             (keep (fn [module]
+                     (when-let [circular-deps (circular-deps module)]
+                       [module circular-deps])))
+             (keys module->deps)))))
```
Comment (crisptrutski) [inline dev/src/dev/deps_graph.clj:94]: Doesn't matter too much, but you might as well unify the 1-arity variant with the `circular-deps` helper as they have the same contract and basically the same name too. The overloaded version will avoid a bunch of necessary calculations, and the code will be a bit less nested too.

Haven't checked this refactor for dumb mistakes.

```suggestion
(defn circular-dependencies
  "Build a graph of module => set of modules it refers to that also refer to this module."
  ([]
   (let [module->deps (module-dependencies)]
     (into (sorted-map)
           (keep (fn [module]
                   (when-let [circular-deps (circular-dependencies module module->deps)]
                     [module circular-deps])))
           (keys module->deps))))
  ([module]
   (circular-dependencies module (module-dependencies)))
  ([module module->deps]
   (letfn [(circular-dependency? [module-x module-y]
             (and (contains? (get module->deps module-x) module-y)
                  (contains? (get module->deps module-y) module-x)))]
     (not-empty (into (sorted-set)
                      (filter (fn [dep]
                                (circular-dependency? module dep)))
                      (get module->deps module))))))
```

File: test/metabase/xrays/test_util/automagic_dashboards.clj [hunk 2]
```diff
@@ -9,7 +9,7 @@
    [metabase.util :as u]
    [metabase.util.malli.schema :as ms]))
 
-(defmacro with-dashboard-cleanup
+(defmacro with-dashboard-cleanup!
```
Comment (crisptrutski) [inline test/metabase/xrays/test_util/automagic_dashboards.clj:12]: Kudos for marking all these helpers

File: src/metabase/api/transform.clj [hunk 3]
```diff
@@ -1,31 +0,0 @@
-(ns metabase.api.transform
```
Comment (crisptrutski) [inline src/metabase/api/transform.clj:1]: Feels good man 🐸

**Review by crisptrutski — APPROVED** [review]

**Comment by camsaul** [comment]
Thanks for the suggestions @crisptrutski, I'll change this stuff in the next PR in the series which has changes to the deps graph stuff anyway

**Pull request merged**